sematic/db/migrate.py

```diff
@@ -47,53 +55,242 @@ def migrate():
+    """
+    Migrate the DB to the latest version.
+    """
+    logging.info("Running migrations on {}".format(get_config().db_url))
+
+    versions = _get_current_versions()
 
     migration_files = _get_migration_files()
 
     for migration_file in migration_files:
         version = migration_file.split("_")[0]
+
         if version in versions:
             logging.info("Already applied {}".format(migration_file))
             continue
 
-        logging.info("Applying {}".format(migration_file))
+        _run_migration(migration_file, version, MigrationDirection.UP)
+
 
-        with open(
-            os.path.join(get_config().migrations_dir, migration_file), "r"
-        ) as file:
-            sql = file.read()
+@main.command("down", short_help="Revert last migration")
+@common_options
+@click.option("--schema-file", "file", type=click.STRING, default="schema.sql")
+def _migrate_down(env: str, verbose: bool, file: str):
+    """
+    Revert the last migration.
+    """
+    _apply_common_options(env, verbose)
+    # Separate function to be able to invoke it outside of click
+    migrate_down()
+    dump_schema(file)
 
-        up_sql = sql.split("-- migrate:down")[0].split("-- migrate:up")[1].strip()
 
-        statements = up_sql.split(";")
+def migrate_down():
+    """
+    Revert the last migration.
+    """
+    current_versions = _get_current_versions()
 
-        with db().get_engine().begin() as conn:
-            for statement in statements:
-                if len(statement) == 0:
-                    continue
+    if len(current_versions) == 0:
+        logging.warning("No migrations to rollback")
+        return
 
-                conn.execute("{};".format(statement))
+    latest_version = current_versions[-1]
 
-            conn.execute(
-                text("INSERT INTO schema_migrations(version) values (:version)"),
-                version=version,
-            )
+    migration_file = next(
+        f for f in _get_migration_files() if f.startswith(latest_version)
+    )
 
+    _run_migration(migration_file, latest_version, MigrationDirection.DOWN)
 
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser("Sematic migration script")
-    parser.add_argument("--env", required=False, default="local", type=str)
-    parser.add_argument("--verbose", required=False, default=False, action="store_true")
-    args = parser.parse_args()
 
-    switch_env(args.env)
+@main.command("dump", short_help="Dump schema")
+@common_options
+@click.option("--schema-file", "file", type=click.STRING, default="schema.sql")
+def _dump_schema(env: str, verbose: bool, file: str):
+    _apply_common_options(env, verbose)
+    dump_schema(file)
 
-    if args.verbose:
-        logging.basicConfig(level=logging.INFO)
 
-    migrate()
+def dump_schema(file: str):
+    parsed_db_url = urlparse(get_config().db_url)
+
+    if parsed_db_url.scheme != "sqlite":
+        logging.warning("Not dumping schema, only SQLite supported at this time")
```

Comment:
Why not postgres? We will need that too

Reply:
We only run PG migrations on the web box when deploying, and we don't commit changes to the schema file. So yes, eventually we need it, but unlike the SQLite schema which is used to create a test db fixture, the PG schema file is never actually used.

Reply:
Should we remove the postgres schema file for now then? So we don't have a confusingly out-of-date schema in the repo? Regradless, we will need to file a ticket for supporting dumping the postgres schema. But just curious--what makes it hard to do?

Reply:
The fractal: retry settings! <n+1> need to update exception schema! <n+1> need python migrations <n+1> need to replace dbmate <n+1> need to support postgres schema dumps from python variant of dbmate 😆

Reply:
> what makes it hard to do?

Probably not hard, but every little added feature adds more work. So I didn't want to get into that here.

Reply:
I removed the pg schema file.